### PR TITLE
Git poller: replace lastRev state instead of updating

### LINKED ISSUE #6100: Git poller state exceeds database field size after some time
When the Git poller covers all branches, it can happen that eventually the `lastRev` state gets too large so that it cannot be stored in the DB anymore (resulting in `Data too long for column 'value_json' at row 1` exceptions). This is because it never forgets branches that used to exist in the past (https://github.com/buildbot/buildbot/blob/master/master/buildbot/changes/gitpoller.py#L239), so its `lastRev` state never shrinks.

I'm wondering whether the `lastRev` state really needs to be updated (and not replaced) by the current set of references. The only reason why update is used (instead of replacing) I can think of is that if a branch is deleted and then restored (with the same HEAD), it will not trigger a change.

I'd like to create a PR which removes the updating, or at least makes it configurable. What is the preferred solution here?

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes #6100.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/changes/gitpoller.py [hunk 1]
```diff
@@ -236,6 +236,8 @@ def poll(self):
                 log.err(_why="trying to poll branch {} of {}".format(
                         branch, self.repourl))
 
+        # replace self.lastRev contents with revs
+        self.lastRev.clear()
         self.lastRev.update(revs)
```
Comment (p12tic) [inline master/buildbot/changes/gitpoller.py:241]: Is there a reason why we can't just set lastRev to revs?

Comment (felixfontein) [reply master/buildbot/changes/gitpoller.py]: Probably not. I've changed it in 01bb6f5269e503bb46ea4288dceadc42e04eff68.

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
Rebased, will merge soon.

**Comment by felixfontein** [comment]
@p12tic thanks a lot for reviewing and merging!

**Pull request merged**